main/store/actions/index.js

```diff
@@ -424,6 +399,12 @@ module.exports = {
   setDappStorage: (u, hash, state) => {
     if (state) u(`main.dapp.storage.${hash}`, () => state)
   },
+  initOrigin: (u, origin, chainId, type) => {
```

Comment:
@goosewobbler just had one idea, can we just create the origin ID once in the ws/http interfaces and add the name of the  origin to the object itself for wherever we need it for display purposes? I think this way will leave a lot less room for potential error down the road if we ever forget to create this UUID before doing a lookup (or do an extra one before calling this method) and it will also remove all these extra calls.

so it would be something like:

interface Origin {
  [uuidKey: string]: {
    chainId: number,
    name: string
}

and then `payload._origin` can always be the id. does this work?

Reply:
I'm actually making some changes to the WS interface so I can make this change

Reply:
ok, np